dpd/p4/sidecar.p4

```diff
@@ -10,14 +10,86 @@ const bit<9> USER_SPACE_SERVICE_PORT = 192;
 #include <constants.p4>
 #include <parser.p4>
 
+// Top-level function to validate IPv4 multicast MAC addresses.
+//
+// Validate the IPv4 multicast MAC address format (RFC 1112, RFC 7042).
+// IPv4 multicast addresses (224.0.0.0/4 or 0xE0000000/4) must
+// use MAC addresses that follow the IANA-assigned OUI format
+// 01:00:5e + 0 as first bit of the second byte,
+// followed by 23 bits derived from the IPv4 address.
+bool validate_ipv4_mcast_mac(
```

Comment:
I think we should consider doing this type of validation in parsing. e.g. if we see a multicast mac when parsing the ethernet header, then we require either seeing seeing an L3 protocol that is tied to multicast macs (like LLDP for example) or a multicast ip address subsequently.

Same goes for v6 validation below.

Reply:
Good point.